src/iface/ethernet.rs

```diff
@@ -179,6 +179,13 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
                                         &eth_frame.src_addr());
                 }
 
+                for raw_socket in sockets.iter_mut().filter_map(Socket::as_raw) {
+                    if let Ok(()) = raw_socket.process(timestamp, &IpRepr::Ipv4(ipv4_repr),
+                                                       ipv4_packet.payload()) {
+                        return Ok(());
```

Comment:
That's not how raw sockets work on hosted systems. To quote Linux man pages:

> When a packet is received, it is passed to any raw sockets which have been bound to its protocol before it is passed to other protocol handlers (e.g., kernel protocol modules).

Note that it does *not* say that the packet is *not* passed to other protocol handlers. Indeed, that's why TCP sockets still work even if you're running tcpdump (or ping) concurrently.